lib/smack/SmackRep.cpp

```diff
@@ -514,14 +516,28 @@ const Stmt* SmackRep::returnValueAnnotation(const CallInst& CI) {
 //
 // }
 
+bool SmackRep::isUnsafeFloatAccess(const Type* elemTy, const Type* resultTy) {
+  bool unsafeFloat = false;
+  if (elemTy->isFloatingPointTy()) {
+    if (!resultTy || (resultTy->isIntegerTy() && resultTy->getIntegerBitWidth() == 8UL)) {
+      if (!SmackOptions::BitPrecise)
```

Comment:
Can't you merge this if condition with the one on line 521 above?

Reply:
Sorry, I do not get it. Can you clarify it?

Reply:
This whole thing can be just one if condition instead of 3 nested ones:
```
if (!SmackOptions::BitPrecise && elemTy->isFloatingPointTy() &&
  (!resultTy || (resultTy->isIntegerTy() && resultTy->getIntegerBitWidth() == 8UL))) {
```

Reply:
I don't think merging the sub-conditions would work because there is an assertion involved. I've rewritten the conditions so that they are more readable.